lib/framework/procbot.ts

```diff
@@ -44,6 +44,42 @@ interface ServiceMap<T> {
  * * Retrieve configuration file from either the local FS or a ServiceEmitter
  */
 export class ProcBot {
+	/**
+	 * A method that recursively dives down a nested structure,
+	 * replacing values flagged <<INJECT_BLAH>> with values from process.env
+	 * @param value  Nested structure to deep dive
+	 * @returns      Identical structure with matching leaf elements replaced
+	 */
+	protected static injectEnvironmentVariables(value: any[] | Object | any): any[] | Object | any {
+		if (_.isArray(value)) {
+			return _.map(value, ProcBot.injectEnvironmentVariables);
+		}
+		if (_.isObject(value)) {
+			return _.mapValues(value, ProcBot.injectEnvironmentVariables);
+		}
+		if (_.isString(value)) {
+			return ProcBot.stringInjector(value);
+		}
+		return value;
+	}
+
+	/**
+	 * A method to inject values from env vars if required
+	 * @param raw  String to translate, use <<INJECT_BLAH>> to retrieve BLAH
+	 * @returns      Environment variable or original string
+	 */
+	private static stringInjector(raw: string): string {
+		if (/^<<INJECT_.*>>$/.test(raw)) {
```

Comment:
Seems a strange way of doing this as opposed to just using a `raw.match()`. 

I think it's also make sense to make this a non-greedy repeating match, as otherwise it's fairly limited and wouldn't be able to deal with strings that use injectables within them:

```
creds: '<<INJECT_USERNAME>>/<<INJECT_PASSWORD>'
```